percy/snapshot.py

```diff
@@ -55,6 +66,45 @@ def fetch_percy_dom():
     response.raise_for_status()
     return response.text
 
+@lru_cache(maxsize=None)
+def get_widths_for_multi_dom(widths):
+    # Deep copy mobile widths otherwise it will get overridden
+    allWidths = fetched_widths.get('mobile', [])[:]
+    if widths and len(widths) != 0:
+        allWidths.extend(widths)
+    else:
+        allWidths.extend(fetched_widths.get('config', []))
+    return list(set(allWidths))
+
+def change_window_dimension(driver, width, height):
+    if CDP_SUPPORT_SELENIUM and driver.capabilities['browserName'] == 'chrome':
+        driver.execute_cdp_cmd('Emulation.setDeviceMetricsOverride', { 'height': height,
+                               'width': width, 'deviceScaleFactor': 1, 'mobile': False })
+    else:
+        driver.set_window_size(width, height)
+
+def capture_responsive_dom(driver, cookies, **kwargs):
+    widths = kwargs.get('widths') or []
+    if 'width' in kwargs:
+        widths = [kwargs.get('width')]
+
+    # cache doesn't work when parameter is a list so passing tuple
+    widths = get_widths_for_multi_dom(tuple(widths))
```

Comment:
refactor this handle widths completely. Idea is to have all the variables declared in single line at the start of functions.